src/Moq/Protected/IProtectedMock.cs

```diff
@@ -38,6 +39,16 @@ public interface IProtectedMock<TMock> : IFluentInterface
 		/// remember to use <see cref="ItExpr"/> rather than <see cref="It"/>.</param>
 		ISetup<TMock> Setup(string voidMethodName, params object[] args);
 
+		/// <summary>
+		/// Specifies a setup for a void method invocation with the given 
+		/// <paramref name="voidMethodName"/>, optionally specifying arguments for the method call.
+		/// </summary>
+		/// <param name="voidMethodName">The name of the void method to be invoked.</param>
+		/// <param name="genericTypeArguments">An array of types to be substituted for the type parameters of the current generic method definition, null for non-generic method.</param>
```

Comment:
> `null for non-generic method`

I'd prefer `null` and an empty `Type[0]` array to be invalid argument values for this new parameter (`genericTypeArguments`). If you want to set up a non-generic method, simply use the existing overloads.

That is, please remove this comment and add a `Guard.NotNull(genericTypeArguments, nameof(genericTypeArguments))` to the new public overloads. (You may of course still use `null` in the internal implementation.)